chargeio/services/Merchant/MerchantService.cs

```diff
@@ -1,22 +1,14 @@
 ﻿using ChargeIO.Infrastructure;
 using ChargeIO.Models;
+using ChargeIO.services;
 
 namespace ChargeIO.Services.Merchant
 {
-	public class MerchantService
+	public class MerchantService : ServiceBase
 	{
-        private string SecretKey { get; }
-        
-        public MerchantService(string secretKey = "")
+        public MerchantService(string secretKey = "") : base(secretKey)

```

Comment:
Inconsistent defaulting to empty string in all services, vs null string in Requestor. I think this should be defaulted to null here, and BaseService should assert (in its constructor logic that defaults the SecretKey) that SecretKey is non-null. At that point, Requestor can assert that SecretKey is non-nullable.

Reply:
I recall running into problems attempting to "simply" add NonNull annotations to some classes while testing. There is no short version of the story so here are a couple of links:

https://stackoverflow.com/questions/30896853/notnull-attribute (note that this conversation is still going as of the end of August)

https://gist.github.com/olmobrutall/31d2abafe0b21b017d56

Reply:
I'm not suggesting marking this as non-null -- I'm referring to the inconsistency of this being defaulted to an empty string here, but defaulted to null in the Requestor methods. We should change the empty string to null here, and modify BaseService can assert that secretKey is non-null in its constructor. This is the same non-null assertion/throw that should be performed in Requestor.PrepareHttpClient() to guard both possible routes.